lib/registry.js

```diff
@@ -165,6 +165,9 @@ class RegistryFetcher extends Fetcher {
         }
         if (this.integrity) {
           mani._integrity = String(this.integrity)
+          if (dist.signatures) {
```

Comment:
to confirm - we only want signatures if we also have integrity?

Reply:
Yes, this was the intent.  We have to have an integrity first for signatures to mean anything.  If there's no integrity that's a separate issue.  When it comes time to validate integrity signatures we should discuss if this is a failure or not.